flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/AbstractChangelogState.java

```diff
@@ -34,10 +34,13 @@
 abstract class AbstractChangelogState<K, N, V, S extends InternalKvState<K, N, V>>
         implements InternalKvState<K, N, V> {
     protected final S delegatedState;
+    protected N currentNamespace;
+    protected KvStateChangeLogger<V, N> changeLogger;
 
-    AbstractChangelogState(S state) {
+    AbstractChangelogState(S state, KvStateChangeLogger<V, N> changeLogger) {
```

Comment:
@Nonnull for KvStateChangeLogger?

Reply:
I slightly prefer to mark nullable fields as such and assume everythin else non-nullable.
But the `notNull` check is missing. Do you think adding it is enough?

Reply:
> I slightly prefer to mark nullable fields as such and assume everythin else non-nullable.
> But `Preconditions.notNull` check is missing. Do you think adding it is enough?

yes, that would also be fine.